src/main.go

```diff
@@ -78,8 +78,10 @@ func main() {
 	connection.Initialize(config.Provider, *endpointsDirectoryPathFlag, "./setup/deployment/raw-code/functions/producer-consumer/api-template.json")
 	if *serverlessDeployment {
 		serverlessDirPath := fmt.Sprintf("setup/deployment/raw-code/serverless/%s/", config.Provider)
-		setup.ProvisionFunctionsServerless(config, serverlessDirPath)
+		setup.ProvisionFunctionsServerless(&config, serverlessDirPath)
+		log.Infof("number of routes %d, numebr of endpoints %d", len(config.SubExperiments[0].Routes), len(config.SubExperiments[0].Endpoints))
 		// TODO: trigger benchmarking.TriggerSubExperiments once implemented (disabled to pass CI pipeline, for now)
+		benchmarking.TriggerSubExperiments(config, outputDirectoryPath, *specificExperimentFlag)
```

Comment:
Let's change the function signature of `benchmarking.TriggerSubExperiments` to 
```diff
- func TriggerSubExperiments(config setup.Configuration, outputDirectoryPath string, specificExperiment int) 
+ func TriggerSubExperiments(config *setup.Configuration, outputDirectoryPath string, specificExperiment int)
```
and change the function call here as well so that we only pass in the pointer for config. It'll be more consistent with other similar functions that way.

Reply:
hey! My reasoning was that if we need to modify the struct we pass the pointer and if we just read it we pass the value. If we decide to always pass the pointer there is a lot more refacoting to be done, as the original code-base passes everyting by the value by default (i.e. every time we pass the subexperiment and such). Do we want to go that way?

Reply:
Okay in that case let's just leave it as is for now 👍